packages/react-native/React/CoreModules/RCTDeviceInfo.mm

```diff
@@ -32,10 +33,34 @@ @implementation RCTDeviceInfo {
   std::atomic<BOOL> _invalidated;
 }
 
+static NSString *const kFrameKeyPath = @"frame";
+
 @synthesize moduleRegistry = _moduleRegistry;
 
 RCT_EXPORT_MODULE()
 
+- (instancetype)init {
+  if (self = [super init]) {
+    UIWindow *mainWindow = RCTKeyWindow();
+    [mainWindow addObserver:self
```

Comment:
What if we do this?
```suggestion
    [RCTKeyWindow() addObserver:self
```
Can we remove the UIKit import? 🤔 

I'm a bit concerned that the internal build system might not like it...

Reply:
Sure let's try removing UIKit import, Im a bit afraid that visionOS won't be happy about it but it can be re-added